### PR TITLE
Share Media and Other Improvements

### PR DESCRIPTION
**Pull request by olokobayusuf**
## Changelog
- #27 
- #22 
- #20 
- #18 
- #13 
- #12

### CODE REVIEW

File: iOS/NatShare/NatShare.m [hunk 1]
```diff
@@ -25,10 +38,15 @@ bool NSShareImage (uint8_t* pngData, int dataSize, const char* message) {
     return true;
 }
 
-bool NSShareVideo (const char* videoPath, const char* message) {
+bool NSShareMedia (const char* videoPath, const char* message) {
```
Comment (cmcfadden) [inline iOS/NatShare/NatShare.m]: Update this to mediaPath so it isn't confusing down the line?

**Comment by cmcfadden** [comment]
I've been playing with this branch in a project that involves taking selfies (ios/android) and then sharing them.  It's *so* much more performant to be able to just toss a jpg to the sharing sheet than dealing with decode/encode and textures.  Really excited to see this approved / merged!

**Comment by olokobayusuf** [comment]
That's great to hear! I want to address #12 and #13 before merging.

**Comment by cmcfadden** [comment]
For what it's worth, on iOS with this branch in place and calling:
`NatShare.Share(path, "My Message Here");`
where path is a path to a JPG in the camera roll, I get "copy to instagram" as one of the share sheet options.  The message itself doesn't pass (I wouldn't expect it to for Instagram) but it does show up.  (Instagram isn't there when I pass a texture)

Still worth doing #12 but that's a nice little bonus of this fix!

**Comment by olokobayusuf** [comment]
#12 tested on iOS in 4f92ea795f47ba0bbbd41b832044cf43579f07c2

**Comment by olokobayusuf** [comment]
Implemented #13. Merging this in a bit.

**Comment by olokobayusuf** [comment]
@cmcfadden Not on this PR because I admin-merged it for urgency; but I've invited you as a collaborator for reviews in the future if you have the time. I'd greatly appreciate it!

**Pull request merged**